lib/src/xdg_secret_portal.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:io';
+
+import 'package:dbus/dbus.dart';
+
+import 'xdg_portal_request.dart';
+
+/// Portal for retrieving application secret.
+class XdgSecretPortal {
+  final DBusRemoteObject _object;
+  final String Function() _generateToken;
+
+  XdgSecretPortal(this._object, this._generateToken);
+
+  /// Get the version of this portal.
+  Future<int> getVersion() => _object
+      .getProperty('org.freedesktop.portal.Secret', 'version',
+          signature: DBusSignature('u'))
+      .then((v) => v.asUint32());
+
+  /// Retrieves a master secret for a sandboxed application.
+  Future<void> retrieveSecret(RandomAccessFile file, {String? token}) async {
```

Comment:
As discussed in the PR, this should return `Future<Uint8List>` and the `RandomAccessFile` be created and read inside this function.